### PR TITLE
Add PR template, README, and contact links

### LINKED ISSUE #2: [Discussion]: Flesh out `FluxML/.github` repository
Originally proposed in https://github.com/FluxML/Zygote.jl/pull/1275#issuecomment-1252811797

This would -
1. add a nice and welcoming README for people visiting `Flux`'s GitHub.
2. allow us to add common files like PR templates, CODE_OF_CONDUCT.md, and maybe issue templates to a central repository.

More on this here - https://docs.github.com/en/communities/setting-up-your-project-for-healthy-contributions/creating-a-default-community-health-file

### PR DESCRIPTION
**Pull request by Saransh-cpp**
Closes #2 

I am not sure if -
- I should add a generic bug report and feature request template to this repository.
- the README should describe `Flux.jl` or `FluxML` as a whole (and how?)

cc: @ToucheSir

### CODE REVIEW

**Comment by ToucheSir** [comment]
1. If project-specific templates can override this, then I'd say go for it.
2. Definitely FluxML. You could borrow some of the description from the website for this.

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 1]
```diff
@@ -0,0 +1,11 @@
+[Please delete this text and describe your change here.
+For bugfixes, please detail the bug and include a test case which your patch fixes.
+If you are adding a new feature, please clearly describe the design, its rationale, the possible alternatives considered.
+It is easiest to merge new features when there is clear precedent in other systems; we need to know we're taking
+the right direction since it can be hard to change later.]
+
+### PR Checklist
+
+- [ ] Tests are added
+- [ ] Entry in NEWS.md
```
Comment (ToucheSir) [inline .github/PULL_REQUEST_TEMPLATE.md:10]:
```suggestion
```
We don't really use this for most repos.

**Comment by Saransh-cpp** [comment]
- Generic bug request template - https://github.com/Saransh-cpp/.github/issues/new?assignees=&labels=bug&template=bug_report.yml
- Generic feature request template - https://github.com/Saransh-cpp/.github/issues/new?assignees=&labels=feature&template=feature_request.yml
- How the "New Issue" page looks like - https://github.com/Saransh-cpp/.github/issues/new/choose
- README file - https://github.com/Saransh-cpp/.github/blob/main/profile/README.md

File: profile/README.md [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+
+## Features
+
+Flux has features that sets it apart among ML systems
+
+### Compiled Eager Code
+Flux provides a single, intuitive way to define models, just like mathematical notation. Julia transparently [compiles your code](https://julialang.org/blog/2018/12/ml-language-compiler), optimising and fusing kernels for the GPU, for the best performance.
+
+### Differentiable Programming
+Existing Julia libraries are differentiable and can be incorporated directly into Flux models. Cutting edge models such as [Neural ODEs](https://julialang.org/blog/2019/01/fluxdiffeq) are first class, and [Zygote](https://github.com/FluxML/Zygote.jl) enables overhead-free gradients.
```
Comment (ToucheSir) [inline profile/README.md:17]: There should be a more up-to-date description of DiffEqFlux over on the SciML site.

File: profile/README.md [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+# The Elegant Machine Learning Stack
+
+Flux is a 100% pure-Julia stack and provides lightweight abstractions on top of Julia's native GPU and AD support. It makes the easy things easy while remaining fully hackable.
+
+## Features
+
+Flux has features that sets it apart among ML systems
+
+### Compiled Eager Code
+Flux provides a single, intuitive way to define models, just like mathematical notation. Julia transparently [compiles your code](https://julialang.org/blog/2018/12/ml-language-compiler), optimising and fusing kernels for the GPU, for the best performance.
```
Comment (ToucheSir) [inline profile/README.md:14]:
```suggestion
Flux provides a single, intuitive way to define models, just like mathematical notation. Julia transparently [compiles your code](https://julialang.org/blog/2018/12/ml-language-compiler), optimizing kernels for the GPU, for the best performance.
```
There's been some confusion around where/when fusion happens before, so let's just avoid opening that can of worms here.

File: profile/README.md [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+
+### First-class GPU support
+GPU kernels can be written directly in Julia via [CUDA.jl](https://github.com/JuliaGPU/CUDA.jl). Flux is uniquely hackable and any part can be tweaked, from GPU code to custom gradients and layers.
+
+### The Model Zoo
+A [rich collection](https://github.com/FluxML/model-zoo) of Flux scripts to learn from, or tweak to your own data. Trained Flux models can be used from [TextAnalysis](https://github.com/JuliaText/TextAnalysis.jl) or [Metalhead](https://github.com/FluxML/Metalhead.jl).
+
+### TPUs & Colab
+Flux models can be [compiled to TPUs](https://github.com/JuliaTPU/XLA.jl) for cloud supercomputing, and run from Google Colab notebooks.
+
```
Comment (ToucheSir) [inline profile/README.md:27]:
```suggestion
```
This never really existed. We should remove it from the site too.

File: profile/README.md [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+Flux provides a single, intuitive way to define models, just like mathematical notation. Julia transparently [compiles your code](https://julialang.org/blog/2018/12/ml-language-compiler), optimising and fusing kernels for the GPU, for the best performance.
+
+### Differentiable Programming
+Existing Julia libraries are differentiable and can be incorporated directly into Flux models. Cutting edge models such as [Neural ODEs](https://julialang.org/blog/2019/01/fluxdiffeq) are first class, and [Zygote](https://github.com/FluxML/Zygote.jl) enables overhead-free gradients.
+
+### First-class GPU support
+GPU kernels can be written directly in Julia via [CUDA.jl](https://github.com/JuliaGPU/CUDA.jl). Flux is uniquely hackable and any part can be tweaked, from GPU code to custom gradients and layers.
+
+### The Model Zoo
+A [rich collection](https://github.com/FluxML/model-zoo) of Flux scripts to learn from, or tweak to your own data. Trained Flux models can be used from [TextAnalysis](https://github.com/JuliaText/TextAnalysis.jl) or [Metalhead](https://github.com/FluxML/Metalhead.jl).
```
Comment (ToucheSir) [inline profile/README.md:23]: At some point we thought about toning down these claims, but I don't remember the specifics. @darsnack do you recall?

Comment (ToucheSir) [reply profile/README.md:23]: That looks good. If we broaden the headline to be more like "Model Ecosystem", the inclusion of Metalhead would make sense as well. I would also nominate Transformers.jl and maybe Flux3D.

File: profile/README.md [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+    <img width="400px" src="https://raw.githubusercontent.com/FluxML/Flux.jl/master/docs/src/assets/logo.png"/>
+</p>
+
+# The Elegant Machine Learning Stack
+
+Flux is a 100% pure-Julia stack and provides lightweight abstractions on top of Julia's native GPU and AD support. It makes the easy things easy while remaining fully hackable.
+
+## Features
+
+Flux has features that sets it apart among ML systems
```
Comment (ToucheSir) [inline profile/README.md:11]: Dangling sentence? Maybe something got lost in a diff.

Comment (Saransh-cpp) [reply profile/README.md:11]: Ah, missed a full stop. This was taken from the website, should this be removed though?

Comment (ToucheSir) [reply profile/README.md:11]: I see. Just the period should be fine.

File: profile/README.md (hunk 5 above)
Comment (darsnack) [reply profile/README.md]: In https://github.com/FluxML/fluxml.github.io/pull/145 I simply removed the line. It's just misleading and drives people to the zoo where the get disappointed. My plan was to put it back in once we port the zoo examples and website tutorials into *real* tutorials that stay up to date in the docs.

I would remove any reference to the zoo here too.

Comment (Saransh-cpp) [reply profile/README.md]: Found it - https://github.com/FluxML/Flux.jl/pull/2069/files#diff-36c852523b6a7513a10b8a6bae21f1037eaa27484efc9042f75bbda46d9f529bR6-R7. Maybe I should add in these  2 points too?

Comment (mcabbott) [reply profile/README.md]: Where does this readme show up?

I wonder if it shouldn't say almost nothing, and direct people to the website / the main Flux repository. 

Having so many different blurbs in different places seems a bit confusing, and harder to keep up to date.

Comment (Saransh-cpp) [reply profile/README.md]: Fair point. This will show up like this - https://github.com/pybamm-team.

**Review by ToucheSir — APPROVED** [review]

**Pull request merged**